### PR TITLE
[FIX] Direct construction of rank data structure from SA without Bwt …

### PR DESCRIPTION
**Pull request by cpockrandt**
…(on secondary memory)

This can now easily be parallelized by adding OMP to the for-loops.

### CODE REVIEW

File: include/seqan/index/index_fm_lf_table.h [hunk 1]
```diff
@@ -630,44 +630,40 @@ _createBwt(LF<StringSet<TText, TSSetSpec>, TSpec, TConfig> & lf, TBwt & bwt, TOt
...
-            *bwtIt = back(text[seqNum - i]);
-            setValue(lf.sentinels, bwtIt - bwtItBeg, false);
+            bwt[i] = back(text[seqNum - (i + 1)]);
+            setValue(lf.sentinels, i, false);
         }
     }
 
     // Compute the rest of the bwt.
-    for (; saIt != saItEnd; ++saIt, ++bwtIt)
+    //#pragma omp parallel for
```
Comment (h-2) [inline include/seqan/index/index_fm_lf_table.h]: this is commented, but the other place isn't. Also, please use SEQAN_OMP_PRAGMA

File: include/seqan/index/index_fm_rank_dictionary_base.h [hunk 2]
```diff
@@ -261,6 +268,91 @@ createRankDictionary(RankDictionary<TValue, TSpec> & dict, TText const & text)
...
+    // Resize the RankDictionary.
+    resize(dict, lengthSum(text) + 1, Exact());
+
+    // Assign the text value by value.
+    TSAIter saItBegin = begin(sa, Standard());
+    TSAIter saItEnd = end(sa, Standard());
+
+    setValue(dict, 0, back(text));
+
+    for (TSAIter saIt = saItBegin; saIt != saItEnd; ++saIt)
```
Comment (h-2) [inline include/seqan/index/index_fm_rank_dictionary_base.h]: this doesnt have parallelization

File: include/seqan/index/index_fm_rank_dictionary_base.h [hunk 3]
```diff
@@ -261,6 +268,91 @@ createRankDictionary(RankDictionary<TValue, TSpec> & dict, TText const & text)
...
+    {
+        if (length(text[seqNum - (i + 1)]) > 0)
+        {
+            setValue(lf.bwt, i, back(text[seqNum - (i + 1)]));
+            setValue(lf.sentinels, i, false);
+        }
+    }
+
+    // Compute the rest of the bwt.
+    #pragma omp parallel for
```
Comment (h-2) [inline include/seqan/index/index_fm_rank_dictionary_base.h]: But this does...

Comment (cpockrandt) [reply include/seqan/index/index_fm_rank_dictionary_base.h]: I was still cleaning up. 😜 You can now merge that code to your lambda. I just left the pragma there so you know where you have to insert it for lambda. I'll remove it before merging this PR.

**Pull request merged**